### ISSUE TITLE
Questions about fine-tuning the Structured3D dataset

### ISSUE THREAD

**Issue by tmzcometolearn** [manycore-research/SpatialLM#136]
Hello, excellent work. I recently tried fine-tuning SpatialLM1.1-Qwen-0.5B on the [structured3d-spatiallm dataset](https://huggingface.co/datasets/ysmao/structured3d-spatiallm), the point cloud used was the unsampled original point cloud. I found that the point cloud in the dataset was downsampled (1cm, 1cm, and 10cm on the x-axis, y-axis, and z-axis, respectively).. After convergence, I tried to inference on some open-source real-world RGBD data (around 100-130㎡), and found that the stability and performance of the inference were not as good as: [SpatialLM1.1-Qwen-0.5B-Structured3D-SFT](https://huggingface.co/ysmao/SpatialLM1.1-Qwen-0.5B-Structured3D-SFT). You mentioned that this SFT model was fine-tuned on the structured3d-spatiallm dataset using [SpatialLM1.1-Qwen-0.5B](https://huggingface.co/manycore-research/SpatialLM1.1-Qwen-0.5B). However, I noticed that its config corresponds to 640 `num_bins`, while the base model of SpatialLM1.1-Qwen-0.5B has 1280 `num_bins`. I would like to ask: during your fine-tuning process, besides changing `num_bins` from 1280 to 640, did you add any other data or perform any data augmentation?

Looking forward to your reply. Best wishes for your work and life.
Thank you~

**Comment by SamMaoYS** [comment]
For structured3d like multiple rooms large scene, we found use 640 bins have better performance for architectural elements detection. So the experiments are conducted in 640 bins for structured3d. If I remembered correctly, no further data augmentation are added in finetuning structured3d. But point jittering, elastic transform can be applied.

**Comment by tmzcometolearn** [comment]
> For structured3d like multiple rooms large scene, we found use 640 bins have better performance for architectural elements detection. So the experiments are conducted in 640 bins for structured3d. If I remembered correctly, no further data augmentation are added in finetuning structured3d. But point jittering, elastic transform can be applied.

Thanks for your reply. I understand that by 'better performance' you mean stability and the rationality of the house generation topology, correct? I haven't conducted relevant ablation experiments myself, but based on my understanding, for llm, predicting geometric structures with 1280 bins is more challenging and less stable compared to 640 bins. So I have two questions:
1. You and your team have performed similar ablation experiments on large multi-room scenes with 1280/640 bins?
2.You mentioned that you fed the model some high‑quality single‑room data and then trained to obtain the model SptialLM-Qwen-0.5B-Layout‑intermd. Would you mind sharing the size (number of samples) of this high‑quality single‑room dataset?
Look forward to your reply.
Thanks a lot.

**Comment by SamMaoYS** [comment]
1. Yes
2. The single rooms are just the spatiallm dataset single rooms, randomly sampled 1~N rooms combination from the same scene when training the room architecture predictions.

**Comment by tmzcometolearn** [comment]
> 1. Yes
> 2. The single rooms are just the spatiallm dataset single rooms, randomly sampled 1~N rooms combination from the same scene when training the room architecture predictions.

https://github.com/manycore-research/SpatialLM/issues/79
In issues 79, you said "Thank you for confirming the issue.
I will investigate this further. One possible cause I can think of is that: the pretraining was done on the SpatialLM dataset, including both single-room and multi-room data. The released checkpoint was finetuned with extra epochs on a subset of higher-quality single-room only data to better optimize for single-room settings. The Structured3D experiment used a checkpoint that was finetuned on the multi-room data with few extra epochs. Once the SpatialLM dataset is released, we will provide more detailed instructions on how to reproduce the experiments for both Structured3D and ScanNet experiments." I would like to confirm the following: 
1.If I need to change [SpatialLM1.1-Qwen-0.5B-Layout-Intmd, 640 bins original](https://huggingface.co/ysmao/SpatialLM1.1-Qwen-0.5B-Layout-Intmd) to num_bins of 1280 and maintain its performance(I want to reduce quantization error), should I add some random 1-N multi-room SpatialLM datasets with [SpatialLM1.1-Qwen-0.5B, 1280 bins](https://huggingface.co/manycore-research/SpatialLM1.1-Qwen-0.5B), then finetune on the Structured3D  Dataset?
2.In the SpatialLM dataset, is the strategy for combining multiple rooms completely random, or does it guarantee that the selected rooms are interconnected?
Look forward to your reply and advice. Best wishes for your work and life.
Thanks.

**Comment by tmzcometolearn** [comment]
> 1. Yes
> 2. The single rooms are just the spatiallm dataset single rooms, randomly sampled 1~N rooms combination from the same scene when training the room architecture predictions.

Hello, could you please take a moment to reply to my question?
Best wishes.

**Comment by SamMaoYS** [comment]
Thank you for the detailed questions.
1. For Structured3D multi-room layout predictions, I suggest stay on 640 bins, and we don't update it to 1280 bins, the 1280 bins is better use for smaller rooms with more finegrained object detection tasks.
2. we simply random sample a subset number of rooms from the same scene, doesn't guarantee the rooms are connected.
